bindings/cpp/iiopp.h

```diff
@@ -309,221 +389,193 @@ class Channel
+{
+    iio_event_stream * const p;
+public:
+
+    EventStream() = delete;
+    EventStream(iio_event_stream * s) : p(s){}
+    operator iio_event_stream * () const {return p;}
+
+    Event read(bool nonblock) {iio_event ev; impl::check(iio_event_stream_read(p, &ev, nonblock), "iio_event_stream_read"); return static_cast<Event&>(ev);} // Flawfinder: ignore
+};
+
+typedef Ptr<EventStream, iio_event_stream, iio_event_stream_destroy> EventStreamPtr;
+
+
 /** @brief C++ wrapper for the @ref Buffer C-API
  */
 class Buffer
 {
     iio_buffer * const p;
 public:
-    Buffer(iio_buffer * buffer) : p(buffer){}
+    Buffer() = delete;
+    Buffer(iio_buffer * buffer) : p(buffer), attrs(buffer){}
     operator iio_buffer * () const {return p;}
-    void cancel() {iio_buffer_cancel(p);}
-    void * end() {return iio_buffer_end(p);}
-    void * first(Channel channel){ return iio_buffer_first(p, channel);}
-    ssize_t for_each(ssize_t (*callback)(const struct iio_channel *chn, void *src, size_t bytes, void *d), void *data){return iio_buffer_foreach_sample(p, callback, data);}
-    void * data() {return iio_buffer_get_data(p);}
+
+#ifndef DOXYGEN
+    typedef impl::AttrSeqT<iio_buffer,
+                           iio_buffer_get_attrs_count,
+                           iio_buffer_get_attr,
+                           iio_buffer_find_attr
+                           > AttrSeq;
+#else
+    typedef impl::AttrSeqT<Buffer> AttrSeq;
+#endif
+
+    AttrSeq attrs;
+
     Device device();
-    int poll_fd() const {
-        int const ret = iio_buffer_get_poll_fd(p);
-        if (ret < 0)
-            impl::err(-ret, "iio_buffer_get_poll_fd");
-        return ret;
-    }
-    size_t push() const { return impl::check_n(iio_buffer_push(p), "iio_buffer_push");}
-    size_t push_partial(size_t samples_count) const {return impl::check_n(iio_buffer_push_partial(p, samples_count), "iio_buffer_push_partial");}
-    size_t refill() const { return impl::check_n(iio_buffer_refill(p), "iio_buffer_refill");}
-    void set_blocking_mode(bool blocking){impl::check(iio_buffer_set_blocking_mode(p, blocking), "iio_buffer_set_blocking_mode");}
+    unsigned int attrs_count() const {return iio_buffer_get_attrs_count(p);}
+    Attr get_attr(unsigned int index) {return iio_buffer_get_attr(p, index);}
+    Attr find_attr(cstr name) {return iio_buffer_find_attr(p, name);}
     void set_data(void * data){iio_buffer_set_data(p, data);}
-    void * start() {return iio_buffer_start(p);}
-    ptrdiff_t step() const {return iio_buffer_step(p);}
+    void * data() {return iio_buffer_get_data(p);}
+    void cancel() {iio_buffer_cancel(p);}
+    void enable() {impl::check(iio_buffer_enable(p), "iio_buffer_enable");}
+    void disable() {impl::check(iio_buffer_disable(p), "iio_buffer_disable");}
+    ChannelsMask channels_mask() {return iio_buffer_get_channels_mask(p);}
+    BlockPtr crate_block(size_t size) { return BlockPtr{impl::check(iio_buffer_create_block(p, size), "iio_buffer_create_block")}; }
```

Comment:
`crate_block` -> `create_block`

Reply:
Oops. Committed a fix.

Reply:
Can you merge it into the first commit maybe?

I'll also need you to sign all the commits (currently only the first one is signed).